frontend/src/metabase/visualizations/lib/LineAreaBarRenderer.js

```diff
@@ -444,20 +444,31 @@ function applyChartLineBarSettings(
   }
 }
 
+const BUBBLE_SIZE_INDEX = 2;
+
+const getBubbleSizeMaxDomain = (datas, seriesIndex) => {
+  const seriesData = datas[seriesIndex];
+  const sizeValues = seriesData.map(data => data[BUBBLE_SIZE_INDEX]);
+  return d3.max(sizeValues);
+};
+
 // TODO - give this a good name when I figure out what it does
```

Comment:
@alxnddr, I'd heartily encourage you to follow up with a PR raising code morale here.

Basic stuff comes to mind:
A fitting function name.
Extracting what happens if `isBubble === true` into its own function.
Renaming the `datas` variable. Plural of plural's been a first for me.

Reply:
@gusaiani thanks for your suggestions

> Extracting what happens if isBubble === true into its own function.

The function has 20 lines and the outer `if` condition which is 19 lines of the function does one particular thing — configures bubble radius. Do you think it worth splitting already not so long function extracting subpart of generally one thing?

> Renaming the datas variable. Plural of plural's been a first for me.

This imperfect name is widely in our codebase:
datas — array of arrays of series data
data — array of series data

It seems like an overkill to rename it in the entire codebase in the scope of a bugfix. What do you think?

> A fitting function name.

Thanks, renamed it

Reply:
Thanks, @alxnddr.

Communication's a hard thing.
My comment wasn't so clear about how important I felt each thought was.

I do think you tackled what I also thought was most important — the function name. Thanks for that.

> Do you think it worth splitting already not so long function extracting subpart of generally one thing?

I personally do think it's worth it.
This isn't the style we use everyday.
One person that summed this idea up well is [Sandi Metz](https://thoughtbot.com/blog/sandi-metz-rules-for-developers).

Another way to do the same thing is by having each function do just one thing, really as much as possible.
As it is, this function does about 4 things.

1. `chart.keyAccessor(d => d.key[0]).valueAccessor(d => d.key[1]);` which is pretty incomprehensible for the non-initiated.

2. Nested if-elses to treat the visuals, with about 3 outcomes including doing nothing.

So extracting at least that longer block, and naming it well (otherwise we may be worse off), in my view reduces cognitive strain. This chunking, imho, changes our lives. We are happier to wake up to work and much more productive.

> It seems like an overkill to rename it in the entire codebase in the scope of a bugfix. What do you think?

Definitely. 

In fact all suggestions were possibilities for future edits.

Reply:
@gusaiani thanks for the clarification. I do personally think that breaking down everything into <=5 lines functions is not a very good pattern and it brings maintainability issues as very long functions do.

First,  the smallest granularity functions that at up to 5 lines makes you jumping to implementation to understand how it works or when you debug it. Compared to that, 20 lines fit in 1/3 of a laptop screen and you don't need to jump to small functions in the same file above or even in different files. So it still seems like to much to break down 20 line functions 4 different tiny functions especially when they are not even extracting any shared reusable logic.
There is a [research](https://softwarebyscience.com/very-short-functions-are-a-code-smell-an-overview-of-the-science-on-function-length/) on this topic.